pkg/openshift/rosa/cluster.go

```diff
@@ -0,0 +1,441 @@
+
+	return clusterID, err
+}
+
+// getCluster gets the cluster the body
+func (r *Provider) getCluster(ctx context.Context, clusterName string) (*clustersmgmtv1.Cluster, error) {
+	query := fmt.Sprintf("product.id = 'rosa' AND name = '%s'", clusterName)
+	response, err := r.ClustersMgmt().V1().Clusters().List().
+		Search(query).
+		Page(1).
+		Size(1).
+		SendContext(ctx)
+
+	switch response.Total() {
+	case 1:
+		return response.Items().Slice()[0], nil
+	default:
+		return nil, fmt.Errorf("cluster %q not found: %v", clusterName, err)
+	}
+}
+
+// deleteCluster handles sending the request to delete the cluster
+func (r *Provider) deleteCluster(ctx context.Context, clusterID string) error {
+	if clusterID == "" {
+		return fmt.Errorf("cluster ID is undefined and is required")
+	}
+
+	log.Printf("Initiate cluster %q deletion\n", clusterID)
+
+	commandArgs := []string{"delete", "cluster", "--cluster", clusterID, "--yes"}
+
+	_, _, err := r.RunCommand(ctx, exec.CommandContext(ctx, r.rosaBinary, commandArgs...))
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Cluster %q deletion initiated!\n", clusterID)
+
+	return err
+}
+
+// waitForClusterToBeInstalled waits for the cluster to be in a ready state
+func (r *Provider) waitForClusterToBeInstalled(ctx context.Context, clusterID string, attempts int) error {
+	getClusterState := func() (string, error) {
+		commandArgs := []string{"describe", "cluster", "--cluster", clusterID, "--output", "json"}
+
+		stdout, _, err := r.RunCommand(ctx, exec.CommandContext(ctx, r.rosaBinary, commandArgs...))
+		if err != nil {
+			return "", err
+		}
+
+		output, err := cmd.ConvertOutputToMap(stdout)
+		if err != nil {
+			return "", fmt.Errorf("failed to convert output to map: %v", err)
+		}
+
+		clusterState := fmt.Sprint(output["status"].(map[string]any)["state"])
+
+		return clusterState, err
+	}
+
+	for i := 1; i <= attempts; i++ {
+		clusterState, err := getClusterState()
+		if err != nil {
+			clusterState = "n/a"
+		}
+
+		if clusterState != "ready" {
+			log.Printf("%d/%d : Cluster %q not in ready state (state=%s)\n", i, attempts, clusterID, clusterState)
+			time.Sleep(1 * time.Minute)
+			continue
+		}
+
+		log.Printf("Cluster id: %q is ready!", clusterID)
+		return nil
+	}
+
+	return fmt.Errorf("cluster %q failed to enter ready state in the alloted attempts", clusterID)
+}
+
+// waitForClusterToBeReady waits for the cluster health check job to succeed
+func (r *Provider) waitForClusterToBeReady(ctx context.Context, kubeconfigFile, reportDir string, hostedCP bool) error {
+	cluster, err := clusterutil.NewFromKubeconfig(kubeconfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to construct openshift client: %v", err)
+	}
+
+	switch hostedCP {
+	case true:
+		return cluster.HCPClusterReadyHealthCheck(ctx)
+	case false:
+		return cluster.OSDClusterReadyHealthCheck(ctx, clusterutil.OSDClusterReadyJobName, reportDir)
+	default:
+		return nil
+	}
```

Comment:
```suggestion
	if hostedCP {
		return cluster.HCPClusterReadyHealthCheck(ctx)
	}
	return cluster.OSDClusterReadyHealthCheck(ctx, clusterutil.OSDClusterReadyJobName, reportDir)
```

Reply:
Resolved in 4116923a7c2c6c8696476a3f25a3551f3bcf3644